Improved Settings Page

GID0317 opened the pull request:
This PR modernizes the Settings UI with Win11-style polish and Discord-inspired profile layout. Features:

- Profile row in Settings navigation with avatar, display name, and search functionality
- Discord-style Accounts summary (vertical banner + overlapping avatar with live presence indicator)
- Display Name / Username ordering (no discriminator tags)
- About page: WinUI-style expander settings card with selectable version
- Notifications/Account summary/Security/Themes: Bundled expanders expander settings card, InfoBar warnings, instant restart dialog
- The OS-aware for specific things between versions. Windows 10 continues to show the Action Center icon along with Live Tiles and People, while Windows 11 uses a Ringer icon, no longer includes Live Tiles and people.
- Show an instant restart dialog when switching a theme and change the "No" button to "Cancel" so that when we click "Cancel," it reselects the previously selected theme.
- Fix the overlapping scrollbar issue on the settings page.
- Move the “Learn More” contrast hyperlink from inside the expander to the descriptions in the settings expander card.

Tested on: Windows 11 Version 25H2 (OS Build 26220.7523) x64

Preview:
- Profile row in Settings navigation with avatar, display name, and search functionality
- Discord-style Accounts summary (vertical banner + overlapping avatar with live presence indicator)
- Display Name / Username ordering (no discriminator tags)
<img width="1920" height="1020" alt="image" src="https://github.com/user-attachments/assets/ea2b3ea8-e721-4d36-bfbc-c6ac929402e5" />
<img width="1920" height="1020" alt="image" src="https://github.com/user-attachments/assets/d27ea1a3-802c-4875-9968-c19f252132d7" />

- About page: WinUI-style expander settings card with selectable version
<img width="1920" height="1020" alt="image" src="https://github.com/user-attachments/assets/b654f451-c8dd-4d37-849b-84c31fa40e97" />

- Notifications/Account summary/Security/Themes: Bundled expanders expander settings card, InfoBar warnings, border separator
<img width="1513" height="575" alt="image" src="https://github.com/user-attachments/assets/53b0695e-538b-4195-9608-ef723b00ea2d" />
<img width="1519" height="516" alt="image" src="https://github.com/user-attachments/assets/c82f267e-8b0d-488b-b902-9cda5bb4cf15" />
<img width="1522" height="678" alt="image" src="https://github.com/user-attachments/assets/a34e6357-2789-487c-928a-c489de4e445c" />
<img width="1487" height="634" alt="image" src="https://github.com/user-attachments/assets/db98089e-8885-4990-a94b-0e2cedf442e5" />

- Instant restart dialog when we are switching a theme
<img width="1919" height="1016" alt="Screenshot 2025-12-27 025406" src="https://github.com/user-attachments/assets/9e905bd2-2529-4ea5-b5d0-d91342a3dbc2" />

- Move the “Learn More” contrast hyperlink from inside the expander to the descriptions in the settings expander card.
<img width="1493" height="251" alt="Screenshot 2025-12-27 025747" src="https://github.com/user-attachments/assets/edc22a73-5320-452e-a5b1-bdc921f26856" />

GID0317 committed:
add profile and autosuggestbox to make similar to latest discord UI

Added profile row in NavigationView pane (avatar + name)
Added AutoSuggestBox search functionality
Compact mode handling (collapsed search, avatar-only profile)
Theme-aware spacing adjustments for SunValley vs Fluent/Performance
Profile name now uses GlobalName instead of discriminator

GID0317 committed:
update the profile card, adapt the depreciated windows 10 related to not shown in windows 11 (Live Tiles, Action Center Symbol), Update the statistic to using expander with copy button

Complete card redesign: vertical banner + overlapping avatar layout
Added users:AvatarControl with presence indicator
Display Name / Username ordering (no discriminator)
Red logout button
Theme-adaptive corner radius
Added UserPresence property with live messenger updates
Statistics moved to expander with "Copy" button

GID0317 committed:
Modernize the about page to use expander

Single expander with Community Toolkit like Settings Expander like-style header
App icon + name + description on left, version on right
Version text selectable, "v" prefix removed
Disclaimer section with border separator
Acknowledgements content properly formatted

GID0317 committed:
Bundled notification toggles under one expander Added OS-adaptive notification glyph (Win11 vs Win10)

Bundled notification toggles under one expander
Added OS-adaptive notification glyph (Win11 vs Win10)

GID0317 committed:
Replace the Windows Hello unavailable warning text with InfoBar

GID0317 committed:
make the restart dialog to immediately appeared when we click cancel it revert to previous selection before change

Immediate restart dialog on theme selection
Cancel reverts theme selection
Removed old "restart required" prompt

GID0317 committed:
Added expander divider between video/audio options

GID0317 committed:
Move "Learn more" moved into description to make it inline with windows 11 design

GID0317 committed:
Update AccountsSettingsModel.cs

Added UserPresence property (PresenceViewModel)
Registered for DiscordEventMessage<PresenceUpdateEventArgs> via messenger
Updates presence indicator in real-time

GID0317 committed:
removed DisplayVersion "v" prefix

GID0317 committed:
Added NotificationGlyph property (OS-aware)

GID0317 committed:
Added AccountsSettings_ProfileCard_CornerRadius: 0 and Added AccountsSettings_ProfileCard_BannerCornerRadius: 0

GID0317 committed:
Added AccountsSettings_ProfileCard_CornerRadius: 8 and Added AccountsSettings_ProfileCard_BannerCornerRadius: 8,8,0,0

GID0317 committed:
Added ExpandableContent property for expandable sections and Icon/content positioning improvements

GID0317 committed:
Layout refinements for Win11 polish

GID0317 committed:
Update AccountsSettingsPage.resw

StatisticsBlock.Title → "Statistics"
StatisticsBlock.Description → "View detailed statistics about your account"
CopyStatisticsButton.Content → "Copy"
CopyStatisticsButtonText.Text → "Copy"

GID0317 committed:
Update SettingsPage.resw

SearchBox.PlaceholderText → "Search"
EditProfileText.Text → "Edit Profiles"

GID0317 committed:
Update SecuritySettingsPage.resw

InfoBar warning text updates
"Learn more" button text

GID0317 committed:
Add dialog cancel string for theme restart dialog

GID0317 committed:
Update Description text for contrast hyperlink

GID0317 committed:
Update NotificationsSettingsPage.resw

Add a OS type related sting like in windows 11 we not specified the changed system and depreciated features but in windows 10 we displayed those

GID0317 committed:
Add similar string like en-US for the settings page

Added and updated several resource strings for Accounts, Messaging, Security, and general Settings pages. Changes include new labels and descriptions for statistics, updated button texts, improved clarity for dialog buttons, and additional accessibility and navigation options.

GID0317 committed:
Add similar string like en-US for the settings page

Added new French translations for statistics and profile editing in settings pages. Updated dialog button text and improved clarity in security settings. These changes enhance localization and user experience for French-speaking users.

GID0317 committed:
Add similar string like en-US for the settings page

Added and refined Japanese resource strings for Accounts, Notifications, Security, and general Settings pages. Changes include new descriptions for Windows 10/11, improved button labels, and additional context for statistics and profile editing.

GID0317 committed:
Add similar string like en-US for the settings page

Added and updated several Russian localization strings in AccountsSettingsPage, Dialogs, SecuritySettingsPage, and SettingsPage. Changes include new descriptions, button texts, and improved wording for clarity and consistency.

GID0317 committed:
Make controls margin inside the expander consistent

GID0317 committed:
Update the banner heigh to similar like discord heigh in the settings

GID0317 committed:
Update the margin  of collapsed mode search icon to correctly centered

GID0317 committed:
Clean up comments in SettingsPage.xaml.cs

Removed redundant and outdated comments

WamWooWam commented on Unicord.Universal/Pages/Settings/SettingsPage.xaml.cs:
```diff
@@ -47,18 +55,53 @@ private static Dictionary<SettingsPageType, Type> _pages
+                {
+                    UserProfilePicture.ProfilePicture = new BitmapImage(new Uri(user.AvatarUrl));
+                }
+            }
```
still on holiday but stuff like this really should be in the ViewModel, this goes for everything across all PRs here, assigning directly to properties on UI elements should be a last resort

GID0317 replied:
Thanks for the feedback! I’ll refactor this into the ViewModel to match project standards and get it patched up soon.

GID0317 committed:
Removed direct UI property assignments

GID0317 committed:
Bound PersonPicture.ProfilePicture and TextBlock.Text to ViewModel

GID0317 committed:
Added ProfileDisplayName and ProfileAvatarUrl properties

GID0317 committed:
Improve account page profile card text styles and layout

Adjusted border thickness and brush for avatar container, increased avatar margin, and updated text styles for display name, username, and email fields to use proper brush style and improved font weights.

GID0317 committed:
Add 'Copy User ID' option under more button to account page settings

GID0317 committed:
Fix indentation in AccountsSettingsPage.xaml
